flfm/app/main.py

```diff
@@ -0,0 +1,470 @@
+                                                "Center x",
+                                                dbc.Input(
+                                                    value=settings.DEFAULT_CENTER_X,
+                                                    type="number",
+                                                    id="center_x",
+                                                    debounce=app_settings.DEBOUNCE,
+                                                ),
+                                                "Center y",
+                                                dbc.Input(
+                                                    value=settings.DEFAULT_CENTER_Y,
+                                                    type="number",
+                                                    id="center_y",
+                                                    debounce=app_settings.DEBOUNCE,
+                                                ),
+                                                "Radius",
+                                                dbc.Input(
+                                                    value=settings.DEFUALT_RADIUS,
+                                                    type="number",
+                                                    id="radius",
+                                                    debounce=app_settings.DEBOUNCE,
+                                                ),
+                                            ]
+                                        )
+                                    ),
+                                    dbc.Col(
+                                        html.Div(
+                                            [
+                                                dbc.Button("Reconstruct", id="run-reconstruction", n_clicks=0),
+                                            ]
+                                        )
+                                    ),
+                                ]
+                            ),
+                            dcc.Loading(
+                                id=dict(type="loading", index="reconstruction"),
+                                type=app_settings.LOADING_TYPE,
+                                delay_show=app_settings.LOADING_DELAY_SHOW,
+                                children=html.Div(id=dict(type="display", index="reconstruction")),
+                            ),
+                        ],
+                    ),
+                ]
+            ),
+        ]
+    )
+    return layout
+
+
+dash_app.layout = dash_layout
+
+
+@callback(
+    Output(dict(type="display", index=MATCH), "children", allow_duplicate=True),
+    Output(dict(type="n-frames", index=MATCH), "data"),
+    Input(dict(type="upload", index=MATCH), "contents"),
+    State(dict(type="upload", index=MATCH), "filename"),
+    State(dict(type="upload", index=MATCH), "id"),
+    prevent_initial_call=True,
+)
+def upload_data(contents, filename, id):
```

Comment:
Can you add some type hints here if the types are consistent and reasonable to reference?